# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1314996521 -3600
#      Fri Sep 02 21:48:41 2011 +0100
# Node ID 12ba32dd345818a58d591098fd0f340ba5f0f75c
# Parent  8afb81b3274833cd98034797e6074df06d4611e9
Mark profiler class and data for export.

* src/profiler: Add OCTINTERP_API modifier to profile_data_accumulator and
profiler.

diff --git a/src/profiler.h b/src/profiler.h
old mode 100644
new mode 100755
--- a/src/profiler.h
+++ b/src/profiler.h
@@ -26,17 +26,17 @@ along with Octave; see the file COPYING.
 #include <map>
 #include <set>
 #include <string>
 #include <vector>
 
 class octave_value;
 
 class
-OCTAVE_API
+OCTINTERP_API
 profile_data_accumulator
 {
 public:
 
   // This is a utility class that can be used to call the enter/exit
   // functions in a manner protected from stack unwinding.
   class enter
   {
@@ -168,17 +168,17 @@ private:
   void add_current_time (void);
 
   // No copying!
   profile_data_accumulator (const profile_data_accumulator&);
   profile_data_accumulator& operator = (const profile_data_accumulator&);
 };
 
 // The instance used.
-extern profile_data_accumulator profiler;
+extern OCTINTERP_API profile_data_accumulator profiler;
 
 // Helper macro to profile a block of code.
 #define BEGIN_PROFILER_BLOCK(name) \
   { \
     profile_data_accumulator::enter pe (profiler, (name));
 #define END_PROFILER_BLOCK \
   }
 
